File: BAT-and-UTIL-files-1/insert-before-each-line.pl
#!/usr/local/bin/perl

#### THE PYTHON VERSION IS 32% SLOWER....BUT IT HANDLES EMOJI AND PERL CAN’T

### substitutes {{{{QUOTE}}}}   into "
### substitutes {{{{QUOTES}}}}  into "
### substitutes {{{{PERCENT}}}} into %
### substitutes {{{{PCT}}}}     into %
### substitutes {{{{PIPE}}}}    into |

use utf8;
binmode(STDIN,  ':encoding(UTF-8)');
binmode(STDOUT, ':encoding(UTF-8)');
use open ':std', ':encoding(UTF-8)';


my $line;
my $arg=$ARGV[0];																																	#DEBUG: #use Encode; @ARGV = map { decode("UTF-8", $_) } @ARGV;#DEBUG: #for my $arg (@ARGV) { print "Raw ARGV: ", join(" ", map { sprintf("\\x%02X", ord($_)) } split(//, $arg)), "\n"; } print "ARGV[0]: $ARGV[0]\n";

$arg =~ s/{{{{QUOTE}}}}/"/g;
$arg =~ s/{{{{QUOTES}}}}/"/g;
$arg =~ s/{{{{PERCENT}}}}/%/g;
$arg =~ s/{{{{PCT}}}}/%/g;
$arg =~ s/{{{{PIPE}}}}/|/g;

while ($line=<STDIN>) {
    chomp $line;
    print $arg . $line . "\n";
}

File: BAT-and-UTIL-files-1/insert-before-each-line.py
#!/usr/bin/env python3

#### THE PERL VERSION IS 25% FASTER! BUT THIS ONE HANDLES EMOJI! ... [and has the percent substitution] ...

### substitutes {{{{QUOTE}}}}   into "
### substitutes {{{{PERCENT}}}} into %

import sys
sys.stdin .reconfigure(encoding='utf-8', errors='replace')
sys.stdout.reconfigure(encoding='utf-8', errors='replace')

def main():
    if len(sys.argv) < 2:
        print(f"Usage: {sys.argv[0]} <prefix>", file=sys.stderr)
        sys.exit(1)
    prefix = sys.argv[1]

    prefix = prefix.replace("{{{{QUOTE}}}}"  , r'"')
    prefix = prefix.replace("{{{{QUOTES}}}}" , r'"')
    prefix = prefix.replace("{{{{PERCENT}}}}", r'%')
    prefix = prefix.replace("{{{{PCT}}}}"    , r'%')
    prefix = prefix.replace("{{{{PIPE}}}}"   , r'|')

    for line in sys.stdin:
        line = line.rstrip('\n')
        # Use a normal string, not a raw string, to ensure correct matching
        print(prefix + line)

if __name__ == "__main__":
    main()
